### DISCUSSION TITLE
CrazyRadio PA Firmware Upgrade Help for crazyflie-link-cpp

### DISCUSSION THREAD

**Discussion by yakuman2800** [bitcraze/discussions#1034] [Q&A] [answered]
I built the crazyflie-link-cpp library, connected my CrazyRadio PA to my computer and tried running "example_broadcast.exe".
https://github.com/bitcraze/crazyflie-link-cpp/blob/master/examples/broadcast.cpp

I got the following error message saying my radio was on firmware 0.83, but when I ran the command to check the version, it gave me 0.53.
![image](https://github.com/bitcraze/discussions/assets/122763402/e97a5f02-c1e6-45cc-a663-240aca5146af)

I know I must upgrade the firmware, so I have the following question:

Which resource's instructions should I follow to do this?
https://www.bitcraze.io/documentation/repository/crazyradio-firmware/master/building/programming/
https://github.com/bitcraze/crazyradio-firmware/blob/master/docs/building/building_flashing.md

And how do I know:
What version I'm installing using these resources?
Whether or not this version I'm installing is sufficient to support broadcast communication?

I'm being extra careful because one of the links says if I do this incorrectly, I risk "bricking the radio".

**Comment by yakuman2800** [comment, marked as answer]
Update: the firmware on your repo is at v99.55 at time of writing, and flashing the crazyradio with this version supports broadcast communication, so this issue is resolved.

**Reply by knmcguire** [reply]
great you found out how to do it. I think that the crazyswarm2 community can also help out more with this as they use crazyflie_link_cpp more regurally. Just a note for the future.
